src/test/java/org/takes/tk/TkProxyTest.java

```diff
@@ -49,23 +50,273 @@
      */
     @Test
     public void justWorks() throws Exception {
-        new FtRemote(new TkFixed("hello, world!")).exec(
+        final String testText = "Hello, world!";
+        new FtRemote(new TkFixed(testText)).exec(
             new FtRemote.Script() {
                 @Override
                 public void exec(final URI home) throws IOException {
                     MatcherAssert.assertThat(
                         new RsPrint(
                             new TkProxy(
-                                String.format(
-                                    "%s:%d",
-                                    home.getHost(), home.getPort()
-                                )
+                                TkProxyTest.this.formatProxyHost(home)
                             ).act(new RqFake())
                         ).print(),
-                        Matchers.containsString("hello")
+                        Matchers.containsString(testText)
+                    );
+                }
+            }
+        );
+    }
+
+    /**
+     * TkProxy can work with GET HTTP method
+     * Plus 4 random headers checked.
+     * @throws Exception If some problem inside
+     */
+    @Test
+    public void actsOnHttpGETWithRequestHeaders() throws Exception {
+        final String testText = "GETTest";
+        new FtRemote(new TkFixed(testText)).exec(
+            new FtRemote.Script() {
+                @Override
+                public void exec(final URI home) throws IOException {
+                    MatcherAssert.assertThat(
+                        TkProxyTest.this.actResponse(
+                            home,
+                            javax.ws.rs.HttpMethod.GET,
```

Comment:
@aschworer is it necessary to put full package name here?
I gave you the full name just to avoid a mess if the same class exists in other packages (usual situation)